lib/pages/login_page/login_page.dart

```diff
@@ -0,0 +1,144 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:mobile_project/custom_widget/common_widgets_lib.dart';
+import 'package:mobile_project/pages/login_page/login_cubit.dart';
+import 'package:mobile_project/pages/login_page/login_state.dart';
+
+class LoginPage extends StatelessWidget {
+  const LoginPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<LoginCubit>(
+      create: (_) => LoginCubit(
+        loginService: context.read(),
+        userService: context.read(),
+        networkService: context.read(),
+      )..checkSavedSession(),
+      child: BlocConsumer<LoginCubit, LoginState>(
+        listener: (context, state) {
+          if (state.autoLoginSuccess || state.loginSuccess) {
+            Navigator.pushReplacementNamed(context, '/home');
+          }
+          if (state.errorMessage != null) {
+            ScaffoldMessenger.of(context).showSnackBar(
+              SnackBar(
+                content: Text(state.errorMessage!),
+                backgroundColor: Colors.red,
+              ),
+            );
+            context.read<LoginCubit>().clearError();
+          }
+        },
+        builder: (context, state) {
+          return Scaffold(
+            appBar: AppBar(
+              title: const Text('Login'),
+              backgroundColor: const Color(0xFFD86FFF),
+              titleTextStyle: const TextStyle(
+                color: Colors.white,
+                fontSize: 26,
+              ),
+            ),
+            body: Center(
+              child: SingleChildScrollView(
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    if (!context.read<LoginCubit>().networkService.isConnected)
```

Comment:
Better to add something like "bool noInternet" in state and update it in cubit

Reply:
done